pkg/generators/forwarding/fluentd/fluent_conf.go

```diff
@@ -113,7 +113,46 @@ func (conf *outputLabelConf) BufferPath() string {
 }
 
 func (conf *outputLabelConf) IsTLS() bool {
-	return url.IsTLSScheme(conf.URL.Scheme) || conf.Secret != nil
+	return url.IsTLSScheme(conf.URL.Scheme)
+}
+
+func (conf *outputLabelConf) HasUsernamePassword() bool {
+	if conf.Secret == nil {
+		return false
+	}
+
+	if _, ok := conf.Secret.Data["username"]; !ok {
+		return false
+	}
+	if _, ok := conf.Secret.Data["password"]; !ok {
+		return false
+	}
+	return true
+}
+
+func (conf *outputLabelConf) HasTLSKeyAndCrt() bool {
+	if conf.Secret == nil {
+		return false
+	}
+
+	if _, ok := conf.Secret.Data["tls.key"]; !ok {
```

Comment:
I believe `constants.go` has a value for this and the other opinionated secrets

Reply:
Fixed